tests/lib_testslide.py

```diff
@@ -334,3 +334,14 @@ def fails_for_wrong_type(self):
     @context.example
     def fails_for_mock_with_wrong_template(self):
         self.assert_fails(StrictMock(template=int))
+
+    @context.example
+    def passes_for_valid_forward_reference(self):
+        class Foo:
+            @classmethod
+            def get_maybe_foo(cls) -> Optional["Foo"]:  # noqa: F821 undefined name Foo
+                return cls()
+
+        testslide.lib._validate_return_type(
```

Comment:
This test is dependent on the extra caller frame info argument, shouldn't we update all _validate_return_type calls everywhere to pass this argument as well?

Reply:
`_validate_return_type` has already the `caller_frame_info` as mandatory argument:
```
def _validate_return_type(template, value, caller_frame_info):
...
```
Can you elaborate?

Reply:
Oh, sorry, nevermind, I confused this with another function, this is fine :-D